dyne/examples.py

```diff
@@ -533,3 +534,158 @@ def h(k, X, with_jacobian=True):
+        ``wx, wy, wz`` - discrette noise
+
+    The measurement model is::
+
+        [mx, my, mz] = C_bn @ [mn, me, md] + [bx, by, bz]
+
+    Where::
+
+        ``C_bn`` - rotation matrix from ``body`` to ``local level`` frame
+        ``mn, me, md`` - magnetic field in ``local level`` frame, componets of `magnetic_field_n`
+        ``bx, by, bz`` - magnetometer biases, components of `mag_bias`
+        ``mx, my, mz`` - magnetometer measurements
+
+    Parameters
+    ----------
+    total_time : float
+        Total time of the simulation.
+    time_step : float
+        Discretization time step.
+    rph_mean: array_like, shape(3,)
+        Mean value of angles in degrees.
+    rph_change_amplitude: float or array_like, shape(3,)
+        Amplitude of angles variation in degrees.
+    rph_change_period: float or array_like, shape(3,)
+        Period of sinusoidal angles change in seconds.
+    rph_change_phase_offset: array_like, shape(3,)
+        Phase offset for sinusoid part in degrees.
+    magnetic_field_n: array_like, shape(3,)
+        Magnetic field in local level frame.
+    mag_bias: array_like, shape(3,)
+        Magnetometer measurements bias.
+    X0 : array_like or None
+        Initial estimator state. If None, generate it randomly from trajectory and `P0`.
+    P0 : array_like, shape (4, 4)
+        Initial covariance matrix.
+    q : array_like, shape (4,)
+        Additive process noise defined in a continuous sense.
+        Standard deviation of the discrete noise is computed as ``q * sqrt(dt)``, where
+        ``dt`` is `time_step`.
+    sigma_measurement : array_like, shape(3,)
+        Standard deviation of measurement noise.
+    rng : None, int or `numpy.random.RandomState`
+        Seed to create or already created RandomState. None (default) corresponds to
+        nondeterministic seeding.
+
+    Returns
+    -------
+    NonlinearProblemExample
+    """
+    rng = check_random_state(rng)
+    n_epochs = np.round(total_time / time_step).astype(int)
+    time = time_step * np.arange(n_epochs)
+    phase = (2 * np.pi * time[:, None] / rph_change_period +
+             np.deg2rad(rph_change_phase_offset))
+    rph = (np.atleast_2d(rph_mean) +
+           np.atleast_2d(rph_change_amplitude) * np.sin(phase))
+    mn, me, md = magnetic_field_n
+
+    q = np.asarray(q)
+    n_noises = np.sum(q > 0)
+    G = np.zeros((4, n_noises))
+    G[q > 0, np.arange(n_noises)] = 1
+    Q = np.diag(q[q > 0] ** 2)
+
+    def f(k, X, W=None, with_jacobian=True):
+        X = np.array(X)
+        X[0] += (rph[k+1, 2] - rph[k, 2])
+        F = np.eye(4, 4)
+        wk = G @ W if W is not None else 0
+        return (X + wk, F, G) if with_jacobian else X + wk
+
+    def h(k, X, with_jacobian=True):
+        rph_deg = np.array(rph[k])
+        rph_deg[2] = X[0]
+        C_nb = Rotation.from_euler('xyz', rph_deg, degrees=True).as_matrix()
+        Z = C_nb.T @ magnetic_field_n + X[1:]
```

Comment:
Can use `apply` of `Rotation` instead of forming matrix explicitly.

Reply:
`apply` is shorter but matrix multiplication is a little bit more clear what it does.

Reply:
Well, I think it's sort of subjective, either way is fine.